### PR TITLE
Update local install instructions and add add more verbose instructions when node versions don't match

### PR DESCRIPTION
**Pull request by gwwar**
I'm happy to pull this out to two separate issues, but wanted to put up a PR while this is fresh in my mind.

While running through local environment setup steps, I ran `bin/install-node-nvm.sh` and got stuck at the Node update step. Locally, if I don't have `v8.11.3` installed (not shown in `nvm ls`), it exits with an error code of 1 and unhelpfully prints the following:
```
$ ./bin/setup-local-env.sh
STATUS: Updating Node..$
```

New suggested output is now:
```
$ ./bin/setup-local-env.sh
WARNING: Node version does not match the latest long term support version. Please run this command to install and use it:
WARNING: nvm install
WARNING: After that, re-run the setup script to continue.
```

Running `nvm install` manually appear to work fine, and reads appropriately from the .nvmrc file, so I'm not sure why that is happening. I've suggested a script update to prompt the user to `nvm install; nvm use;` since we exit out anyway after the install. Let me know if other folks can reproduce this.

Because I ran into this issue, I didn't realize that docker-compose would be triggered by `bin/install-node-nvm.sh`, until I read through the script. I've added a few README suggestions to clarify what the setup script does, and what it looks like when it runs successfully.

### CODE REVIEW

**Comment by gziolo** [comment]
> Running nvm install manually appear to work fine, and reads appropriately from the .nvmrc file, so I'm not sure why that is happening. I've suggested a script update to prompt the user to nvm install; nvm use; since we exit out anyway after the install. Let me know if other folks can reproduce this.

I think I encountered it two or three times. I agree that the proposed solution is okey. We can always refine it. @pento, any thoughts?

**Review by gziolo — APPROVED** [review]

**Comment by gziolo** [comment]
@gwwar for adding improvements to our docs. I'm sure they are going to be very helpful 👍

**Comment by ntwb** [comment]
I just tested this as I didn't have Node.js `8.11.3` installed, only `8.11.2`:

screen shot 2018-07-10 at 3 06 50 pm

So that pretty much worked for me per the current implementation.

I'm happy to see the docs added, but I don't think we need the actually `nvm` script changes 🤔

File: bin/install-node-nvm.sh [hunk 1]
```diff
@@ -59,12 +59,8 @@ fi
...
 # Check if the current node version is up to date.
 if [ "$TRAVIS" != "true" ] && [ "$(nvm current)" != "$(nvm version-remote --lts)" ]; then
-	echo -en $(status_message "Updating Node..." )
-	nvm install >/dev/null 2>&1
-	echo ' done!'
-
-	echo -e $(warning_message "A new node version was installed, please run this command to use it:" )
-	echo -e $(warning_message "$(action_format "nvm use")" )
+	echo -e $(warning_message "Node version does not match the latest long term support version. Please run this command to install and use it:" )
+	echo -e $(warning_message "$(action_format "nvm install; nvm use")" )
```
Comment (pento) [inline bin/install-node-nvm.sh]: `nvm install` will automatically switch to the correct version, there's no need for the `nvm use`.

**Review by pento — APPROVED** [review]
I'm fine with this change: it doesn't really change the flow of the script, but I'm not particularly surprised that there might be weird environment differences in the shell, vs in the install script.

Left one minor comment, but otherwise, :shipit:.

File: bin/install-node-nvm.sh (hunk 1 above)
Comment (gwwar) [reply bin/install-node-nvm.sh]: Thanks @pento, I've updated that in a5d5f27

**Comment by gwwar** [comment]
Thanks for the reviews!

> I'm happy to see the docs added, but I don't think we need the actually nvm script changes 

@ntwb mainly it was exiting early for me on `nvm install` in the script environment without a prompt on what to do next. I may have something funny in my local env, but didn't want to dig too deeply into it. Since we exit anyway, the change here prompts users to switch and install Node versions instead of attempting to do so automatically in the script

**Pull request merged**